### PR TITLE
possibility to init with 2D array

### LINKED ISSUE #88: Create ukis_pysat.raster-Image with 2D array
**Is your feature request related to a problem? Please describe.**

- `ukis_pysat.raster` only treats 3D arrays. Numerous extermal methods work on a 2D basis and using `np.squeeze()` or `np.expand_dims()` adds unnecessary code.

- building an `ukis_pysat.raster.Image()` from a 2D array will fail with a misleading error message:
  > rasterio._err.CPLE_FileIOError: Free disk space available is 68524220416 bytes, whereas 413266544908 are at least necessary.
  
  We could narrow this problem down to the fact, that bringing a 2D array into a 3D method is simply not working.

**Describe the solution you'd like**
Allow initalization of `ukis_pysat.raster.Image` with both 2D or 3D arrays. If I'm not mistaken, rasterio is limited to 3D arrays. @MWieland and @fwfichtner what is your opinion on this?

### PR DESCRIPTION
**Pull request by fwfichtner**
Fixes #88 

Currently we have no other chance than expanding dims internally. If this is not wanted, don't merge and we have to find another solution.

### CODE REVIEW

**Review by MWieland — APPROVED** [review]
For me it is perfectly fine to expand dims internally if required. Looks like a simple and clean solution to me. Suggest merging.

File: ukis_pysat/raster.py [hunk 1]
```diff
@@ -195,13 +195,15 @@ def __update_dataset(self, crs, transform, nodata=None):
...
             "height": self.__arr.shape[-2],
             "width": self.__arr.shape[-1],
-            "count": self.__arr.shape[0],
+            "count": 1 if self.__arr.ndim == 2 else self.__arr.shape[0],
             "crs": crs,
             "transform": transform,
         }
 
         memfile = MemoryFile()
         with memfile.open(**meta) as ds:
```
Comment (MWieland) [inline ukis_pysat/raster.py:204]: I was just wondering why we don't apply the check [here](https://github.com/dlr-eoc/ukis-pysat/blob/a8bbb98b586bc2429a85dbf0084c2619ee437838/ukis_pysat/raster.py#L55) already where we also check for dimorder, transform and crs? At a first look, this seems more consistent and possibly simpler.

Comment (fwfichtner) [reply ukis_pysat/raster.py:204]: We only need it here, when this method is not called there's no need to do it. Your approach is less hidden though.

Comment (MWieland) [reply ukis_pysat/raster.py:204]: Both options are fine I guess and I would leave it up to you to decide where we should put this. Another benefit of doing the check earlier together with the dimorder check would be that we only do this once and not every time when __update_dataset is called.

**Review by BoehnkeC — APPROVED** [review]
Well done.

File: ukis_pysat/raster.py (hunk 1 above)
Comment (fwfichtner) [reply ukis_pysat/raster.py:204]: I agree, this would make it more consistent because we would always return a 3D array.

**Pull request merged**